### PR TITLE
Add tests for bulk ingest bad documents, more edge case guarding

### PR DESCRIPTION
**Pull request by bryanlb**
###  Summary

This expands the hotfixes done in #972 and #970 to further improve scenarios where documents are sent with malformed data.

### CODE REVIEW

File: astra/src/main/java/com/slack/astra/bulkIngestApi/opensearch/BulkApiRequestParser.java [hunk 1]
```diff
@@ -72,13 +71,25 @@ public static Trace.Span fromIngestDocument(
 
     // See https://blog.mikemccandless.com/2014/05/choosing-fast-unique-identifier-uuid.html on how
     // to improve this
-    String id =
-        Optional.ofNullable(sourceAndMetadata.get(IngestDocument.Metadata.ID.getFieldName()))
-            .map(String::valueOf)
-            .orElse(UUID.randomUUID().toString());
+    String id = UUID.randomUUID().toString();
```
Comment (vthacker) [inline astra/src/main/java/com/slack/astra/bulkIngestApi/opensearch/BulkApiRequestParser.java:74]: calling `UUID.randomUUID()` could be expensive? Perhaps we should only do it if id is not present?

Comment (bryanlb) [reply astra/src/main/java/com/slack/astra/bulkIngestApi/opensearch/BulkApiRequestParser.java]: I think this ends up being the default case, but I've went ahead and moved it to only set if not provided.

**Review by vthacker — APPROVED** [review]
